calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/board_detections/board_detection.py

```diff
@@ -108,33 +107,26 @@ def get_center_2d(self) -> np.array:
         self._cached_center_2d = self.get_flattened_image_points().mean(axis=0)
         return self._cached_center_2d
 
-    def get_reprojection_errors(self, model: Optional[CameraModel] = None) -> np.array:
+    def get_reprojection_errors(self, model: CameraModel) -> np.array:
         """Return the error of projecting the object points into the image and comparing them with the detections."""
-        if model is None:
-            model = self._get_cached_model()
-
-        if (
-            self._cached_camera_model is not None
-            and model == self._cached_camera_model
-            and self._cached_reprojection_errors is not None
-        ):
+        if self._cached_camera_model is None:
```

Comment:
Even if we do not apply the full behavior regarding the 3d statistics that require a non-single shot calibration, the following should still be implemented in this PR:

Have:
`get_reprojection_errors`: uses an existing model
and
`get_single_shot_reprojection_errors`: forces calibration using a single sample. It still requires an input model to know which kind of model it should use (parameters ,etc)

Alternatively, use only `get_reprojection_errors` and implement the required behavior in the caller.

Currently, there are two kind of uses for `self._cached_camera_model`. Maybe the should be split.

Reply:
In this method, what happens if:
`self._cached_camera_model is not None` and `model != self._cached_camera_model` ?

Reply:
> get_reprojection_errors: uses an existing model

Using existing model seems working here after changes in camera model (using last camera matrix & coefficients), but after triggering calibration we obtain bad results. There are some assumptions in `_calibrate` within calibrator and mentioned change in camera model is breaking change.  I would need to reconsider design here again.

Instead of refactor twice, I would prefer to address this issue with [this](https://github.com/tier4/CalibrationTools/pull/208#issuecomment-2487368546) in one shot if it is not a problem.

> In this method, what happens if:
> self._cached_camera_model is not None and model != self._cached_camera_model ?

Model has been changed, so we want to update our reprojection error. Isn't it expected behavior? We can change it.